gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java

```diff
@@ -219,6 +219,33 @@ public void createBucket(String bucketName, CreateBucketOptions options) throws
     }
   }
 
+  /**
+   * See {@link GoogleCloudStorage#copy(String, List, String, List)} for details about expected
+   * behavior.
+   */
+  @Override
+  public void copy(
```

Comment:
Are we dong more than adding UTs in this PR?

Reply:
There are some APIs which share some common code between `GoogleCloudStorageImpl` and `GoogleCloudStorageClientImpl`. For example, this copy method implementation is similar for both the clients -they in turn call the https://github.com/GoogleCloudDataproc/hadoop-connectors/pull/1095/files#diff-23a518b24b3819c111aacec5ec168906787ef291585a52e128c1ce776b8ed0fbR246 which contains client specific implementation. To reduce code duplication, this code was not  added earlier. However, while writing the UTs realised that with the delegator in place, this method will always call the GoogleCloudStorageImpl copy. So added these implementations here as well.

Reply:
Do you want to update the PR description accordingly.

Reply:
Done